connectors/src/connectors/zendesk/temporal/workflows.ts

```diff
@@ -161,18 +170,21 @@ export async function zendeskSyncWorkflow({
   while (categoryIds.size > 0) {
     const categoryIdsToProcess = new Set(categoryIds);
     for (const categoryId of categoryIdsToProcess) {
-      if (!categoryIds.has(categoryId)) {
-        continue;
-      }
       const relatedSignal = categorySignals.find(
-        (signal) => signal.zendeskId === categoryId
+        (signal) => signal.categoryId === categoryId
       );
       const forceResync = relatedSignal?.forceResync || false;
+      const brandId = categoryBrands[categoryId];
+      if (!brandId) {
+        continue; // this is mostly for typing since we add to categoryBrands whenever we add to categoryIds
```

Comment:
nit: better to throw `unreachable` or something so that if someone breaks this code it doesn't just fail silently